### DISCUSSION TITLE
1.29.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#15851] [General]
# Features
- #10003 Allow to temporarily "disable" devices ([`disabled` option](https://www.zigbee2mqtt.io/guide/configuration/devices-groups.html)). Useful for Xmas devices which you are going to re-use next year. 🎄🎅 (@Koenkk)

# Improvements
- #14136 Support HTTPS for frontend server ([docs](https://www.zigbee2mqtt.io/guide/configuration/frontend.html)) (@pheelti)
- #15699 Support Hue effects (candle, fireplace, colorloop) (@sjorge)
- #15120 Improve TuYa TS110E_2gang support (@koenkk)
- [#5043](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5043) Support more features for Sinopé TH1123ZB, TH1124ZB, SP2600ZB (@MattL0)
- [#5067](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5067) Support OTA for GL-S-006P (@Electry)
- [#5031](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5031) Support more features for Sinope TH1400ZB (@faucherc) 
- [#5086](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5086) Change Sinope TH1400ZB `occupied_heating_setpoint` step size to `0.5` (@faucherc)
- [#5088](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5088) Fix SmartThings IM6001-WLP01 battery % accuracy (@sjorge) 
- #15279 Support OTA for Eco-Dim.07/Eco-Dim.10 (@Kilill)
- [#5093](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5093) Support electrical measurements for Third Reality 3RSP019BZ (@weihuan1111)
- [#5102](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5102) Support more features for Sinope TH1123-G2 (@MattL0)
- #15243 Support setting individual gradient colour for Philips device supporting gradient (@zegl)
- [#5113](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5113) Support calibration and precision options for TuYa devices using new integration (@koenkk)
- [#5128](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5128) Expose `current` for TuYa TS000F_power (@vladyspavlov)
- [#5129](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5129) Expose `battery` and `battery_voltage` for TuYa IH012-RT01 (@vladyspavlov)
- #15339  Support backlight mode for TuYa TS0014 (@Koenkk)
- [#5108](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5108) Support `energy` for Ubisys S1, S1-R, S2, D1 and J1 and `action` for Ubisys D1 (@Danit2)
- [#5141](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5141) Expose `battery_voltage` for TuYa TS0210 (@vladyspavlov)
- [#5144](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5144) Expose `power_apparent` for Legrand 067775/741811 (@Mephistofeles)
- [#5137](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5137) Support more features for Xiaomi WS-USC01 and WS-USC02 (@CarlosGomez-dev)
- [#5151](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5151) Support OTA for Develco MOSZB-140 (@sjorge)
- [#5156](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5156) Add OTA support for Xiaomi `WXKG13LM` (@Otnow)
- [#5150](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5150) Support `pi_heating_demand` and reporting of various attributes for Bosch BTH-RA (@BBJake)
- #15568 Fix unknown power source of Xiaomi DLKZMK11LM (@Koenkk)
- [#5161](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5161) Add OTA support for some Namron devices (@atus)
- #15574  Support calibrate and precision options for more attributes of Xiaomi and electrical measurement devices (@Koenkk)
- [#5104](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5104) Support more features for Xiaomi ZNCLBL01LM (@mandrean)
- #15569  Support `switch_type` and `power_on_behavior` for TuYa TS110E_1gang_1 (@Koenkk)
- #15549 Update Paulmann 501.34 actions (@okastl)
- #15502 Use fileVersion for Home Assistant update sensor instead of softwareBuildID (@Koenkk)
- #15555 Expose cover `moving` state to Home Assistant (@cmiguelcabral)
- #5152 Allow reading `power` and `energy` for Ubisys devices (@sjorge)
- #15619  Support setting cover `position` with `scene_add` (@Koenkk)
- #15684 Support `child_lock` for TuYa TS011F_2_gang_wall (@Koenkk)
- [#5186](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5186) Support OTA for Legrand 064882 (@EXP-Carthage)
- [#5191](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5191) Support `temperature` for Visonic MCT-350 SMA (@alsobrsp)
- #15669 Support `backlight_mode` for TuYa TS0012 (@Koenkk)
- #15733 Reduce Docker image size by +- 48% (@Koenkk)
- #15643  Add `factory_reset` for all devices (@Koenkk)
- [#5149](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5149) Support `power` for Vimar 14593 (@mattiazeni)
- [#5207](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5207) Add `off` system mode and improve `current_heating_setpoint` range for Moes BRT-100-TRV (@tefracky)
- #15310 Support `energy` for HEIMAN HS2SK (@geelenbert)
- [#5216](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5216) Support `led_on_motion` for Datek HSE2927E (@TheStigh)
- [#5220](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5220) Specify Moes BRT-100-TRV `boost_heating_countdown_time_set` step value  (@tefracky)
- #15796 Provide `linux/386` Docker image (@Koenkk)
- #15770 Keep cached device state until Zigbee2MQTT is stopped (@MatAtBread)
- [#216](https://github.com/Koenkk/zigbee-OTA/pull/216) Support OTA for Namron 4512705 (@jomaskm)

# Fixes
- [#5096](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5096) Fix TuYa TV02-Zigbee behaviour (@vladi1234)
- [#1000](https://github.com/Koenkk/zigbee-herdsman-converters/issues/1000) #15328 Fix OTA update downloads failing (@koenkk)
- [#5059](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5059) Remove unsupported `tamper` from Sonoff SNZB-03 (@kbickar)
- #5052  Fix no `energy` reported for TuYa TS011F_plug (@Koenkk)
- [#5077](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5077) Detect `_TZ3000_qeuvnohg` as TS011F_din_smart_relay (@frayos)
- [#14541](https://github.com/Koenkk/zigbee2mqtt/issues/14541) Fix gangs not controlled individually for TuYa TS0013_switch_module (@koenkk)
- [#5081](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5081) Fix Avatto ME167 thermostat (`_TZE200_bvu2wnxz`) support (@vladyspavlov)
- [#5090](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5090) Fix Sinope TH1300ZB max `occupied_heating_setpoint` and `unoccupied_heating_setpoint` value (@MattL0)
- [#5103](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5103) Fix `pi_heating_demand` and `running_state` publishing for Sinope thermostats (@MattL0)
- #14607 Fix `state` for Zemismart ZM-CSW032-D (@Koenkk)
- [#5121](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5121) Fix `humidity` precision for TuYa TS0201 (`_TZ3000_ywagc4rj`) (@Koenkk)
- [#5119](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5119) Remove unsupported `power_on_behavior` from Moes ZLD-RCW (@Hadatko)
- #15386 Fix Zemismart TB25 not controllable (@Koenkk)
- [#5124](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5124) IKEA TRADFRI LED1925G6 turns off at brightness 1 (@Electry) 
- #15299  Fix failing configure blocking startup (@Koenkk)
- #14361 Fix Yookee D10110 cover state (@Koenkk)
- #15546 Fix unknown power source of Xiaomi LLKZMK11LM (@Koenkk)
- #15554  Disable unsupported `power_on_behavior` for Lidl HG06106C and Ledvance AC25697 (@Koenkk)
- #15563 Fix error when power is received from Datek HSE2905E (@Koenkk)
- [#5158](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5158) Disabled unsupported `power_on_behaviour` for TuYa TS050XX devices. (@Hadatko)
- #4378  Fix incorrect `current` and `power` values for Centralite 4257050-RZHAC (@Koenkk)
- #15101 Fix Legrand 067776 cover state (@Koenkk)
- [#5190](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5190) Fix Sinope TH1123ZB-G2 group set/get (@MattL0)
- #15709 Remove unsupported `tamper` from Iris 3326-L and 3320-L (@Koenkk)
- [#5200](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5200) Fix Yale ZYA-C4-MOD-S lock/unlock state (@ChipWolf)
- #15741 Fix extension path to make debugging easier in VS Code (@MatAtBread)
- #15191 Fix `temperature` value of TuYa TS0601_soil (@kris969)
- #15767 Fix EcoDim ED-10015 battery percentage (@Koenkk)
- [#5219](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5219) Fix OTA updates with new Hue OTA images (@pedrolamas)
- [#5225](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5225) Fix Develco SMSZB-120 alarm with 4.0.5 firmware (@TheStigh)
- [#5232](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5232) Detect `_TZ3000_6ygjfyll` as TuYa IH012-RT01 (@x-magic)
- [#5231](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5231) Fix  'INSUFFICIENT_SPACE' config error Sinopé TH1123ZB-G2 (@MattL0)

# New supported devices
This release adds support for 60 devices:
- `067776_001a` Legrand Netatmo wired shutter switch (@koenkk)
- `07505L` Immax Neo smart keypad (@JiriBalcar)
- `3315-Geu` Iris Water sensor (@ExMacro)
- `3802968` Namron LED Filament Flex 5W CCT E27 Clear (@TheStigh)
- `3RSP02028BZ` Third Reality Zigbee / BLE smart plug with power (@weihuan1111)
- `4052899930377` OSRAM Lightify pro push button controller (PBC) (@danieledwardgeorgehitchcock)
- `4058075729360` LEDVANCE SMART+ Outdoor Flex multicolor RGBW (@dhanukapulkit)
- `4058075729384` LEDVANCE SMART+ Indoor Flex multicolor RGBW (@dhanukapulkit)
- `4512735` Namron Multiprise with 4 AC outlets and 2 USB super charging ports (16A) (@roysve)
- `5047130P6` Philips Hue White ambiance Buckram single spotlight with bluetooth (@Koenkk)
- `7602031K6` Philips Hue Go portable light (@Seoulcialite)
- `8719514338524` Philips Hue white ambiance Pillar spotlight with Bluetooth (black) (@danieledwardgeorgehitchcock)
- `8750001213` Bosch Twinguard (@safakaltun)
- `915005987301` Philips Hue Gradient Signe table lamp (white) (@jvmiert)
- `915005987501` Philips Hue Gradient Signe floor lamp (white) (@Koenkk)
- `929002376803` Philips Hue Iris copper special edition (generation 4) (@lbarnkow)
- `929002469216` Philips Hue white E27 1100lm with Bluetooth (@danieledwardgeorgehitchcock)
- `929003021301` Philips Hue White E17 470 lumen (@danieledwardgeorgehitchcock)
- `9290030514` Philips Hue Filament Standard A60/E27 bluetooth (@xxavarlonxx)
- `929003053201` Philips Hue Liane (white) (@zegl)
- `929003055401` Philips Hue white ambiance Still (aluminium) (@rikmueller)
- `929003089301` Philips Hue White and Color Ambiance Lucca wall light (@blubaru14)
- `929003098901` Philips Hue White Inara outdoor light (@Koenkk)
- `929003526301` Philips Hue White & Color Ambiance Xamento M (@danieledwardgeorgehitchcock)
- `AU-A1ZBMPRO1ZX` Aurora Lighting AOne MPROZX fixed IP65 fire rated smart tuneable LED downlight (@fatmcgav)
- `B07CVL9SZF` Quotra Dimmer (@sktaylortrash)
- `B07JHL6DRV` Quotra RGB WW LED strip (@sktaylortrash)
- `CK-BL702-AL-01` TuYa Zigbee LED bulb (@Koenkk)
- `DG3HL-1BW` Leviton Indoor Decora smart Zigbee 3.0 certified plug-in dimmer (@brendanpward)
- `DMZ250` Wisdom Zigbee led dimmer 5-250 Watt (@SjonHortensius)
- `HSE2920E` Datek Door/window sensor (@TheStigh)
- `IST-CDZFB2AS007NA-MZN-02` ADEO ENKI LEXMAN E27 LED Edison white filament 806 lumen (@lalexdotcom)
- `KQJCMB11LM` Xiaomi Aqara air monitoring panel S1 (@avdept)
- `RC 210` Innr Remote control (@echocrow)
- `RM3500ZB` Sinopé Calypso smart water heater controller (@ticed35)
- `SA-030-1` eWeLink Zigbee 3.0 smart plug 13A (3120W)(UK version) (@jamesmelville)
- `SNZB-02D` SONOFF Temperature and humidity sensor with screen (@guozi7788)
- `SR-ZG9010A` Sunricher Door windows sensor (@asgothian)
- `SSWQD02LM` Xiaomi Aqara smart dimmer controller t1 pro (@youknowone)
- `SSWQD03LM` Xiaomi Aqara spotlight T2 (@youknowone)
- `SWITCH-ZR03-1` eWeLink Zigbee smart switch (@bmurphy1976)
- `TH1123ZB-G2` Sinopé Zigbee line volt thermostat (@MattL0)
- `TS0026` TuYa 6 button scene wall switch (@evangelo18)
- `TS0046` TuYa Wireless switch with 6 buttons (@kamms)
- `TS0202_3` TuYa Motion detector with illuminance (@fulup-bzh)
- `TS0601_clamp_meter` TuYa Clamp meter (@Koenkk)
- `TS0601_dimmer_knob` TuYa Zigbee smart knob dimmer (@tefracky)
- `TS0601_gas_sensor_2` TuYa Gas sensor (@Masterz69)
- `TS0601_smoke_2` TuYa Photoelectric smoke detector (@Jonno12345)
- `TS0601_soil` TuYa Soil sensor (@koenkk)
- `TS0601_thermostat_3` TuYa Thermostatic radiator valve (@vladyspavlov)
- `TS110E_1gang_2` TuYa 1 channel dimmer (@koenkk)
- `VA4220ZB` Sinopé Sedna smart water valve (@MattL0)
- `WDE002961` Schneider Electric Rotary dimmer (@andredahlqvist)
- `WISZB-137` Develco Vibration sensor (@dlq84)
- `WLCKG1` Wyze Lock (@salman2learn)
- `ZB-SW05` eWeLink Smart light switch - 5 gang (@jwg-123)
- `ZB02A` Netvox Wireless wall switch (@hobbyquaker)
- `ZNCLDJ14LM` Xiaomi Aqara C2 curtain motor (@deiger)
- `ZTH01` TuYa Temperature and humidity sensor (@Burki24)

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.29.0'>1.29.0</a>.</em>

**Comment by ExMacro** [comment]
Thanks for the update @Koenkk ! It seems I managed to mess something up with the https://www.zigbee2mqtt.io/devices/3315-Geu.html update. It's working ok but for some reason in the documentation it says "Iris" as manufacturer, should be "Centralite". Zigbee2mqtt is showing everything correctly but in this list and the docs the manufacturer is wrong. I probably messed up with something when posting files.. sorry about that.

**Reply by Koenkk** [reply]
Fixed! Will be included in the next release on 1 February.

**Reply by ExMacro** [reply]
Thank you very much!

**Comment by svengreb** [comment]
First things first: Happy new year! 🎉🎇
The `disabled` setting and your example use case in this changelog is the reason I searched through the backlog 😀
Disabling my light bulb of my "christmas star", while it rests in a box in the basement the rest of the year, is why I'll update right now.
Thanks for this amazing project! It powers almost all of my smart 🏡 setup 🐝

**Comment by fireRoads** [comment]
After update from 1.28.4-1 to 1.29.0-1 zigbee2mqtt fails to start. Seems to be related to my external converter files since it starts if I remove the external converters from configuration.yaml. I cannot find anything strange with my converter files and they work perfectly with 1.28.4-1. Have you changed something regarding this?

This is the external converter file I have problems with:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const e = exposes.presets;
const ea = exposes.access;

const hueExtend = {
    light_onoff_brightness: (options={}) => ({
        ...extend.light_onoff_brightness(options),
        toZigbee: extend.light_onoff_brightness(options).toZigbee.concat([tz.hue_power_on_behavior, tz.hue_power_on_error]),
    }),
    light_onoff_brightness_colortemp: (options={}) => ({
        ...extend.light_onoff_brightness_colortemp(options),
        toZigbee: extend.light_onoff_brightness_colortemp(options).toZigbee.concat([tz.hue_power_on_behavior, tz.hue_power_on_error]),
    }),
    light_onoff_brightness_color: (options={}) => ({
        ...extend.light_onoff_brightness_color({supportsHS: true, ...options}),
        toZigbee: extend.light_onoff_brightness_color({supportsHS: true, ...options}).toZigbee
            .concat([tz.hue_power_on_behavior, tz.hue_power_on_error]),
    }),
    light_onoff_brightness_colortemp_color: (options={}) => ({
        ...extend.light_onoff_brightness_colortemp_color({supportsHS: true, ...options}),
        toZigbee: extend.light_onoff_brightness_colortemp_color({supportsHS: true, ...options})
            .toZigbee.concat([tz.hue_power_on_behavior, tz.hue_power_on_error]),
    }),
};

const definition = {
    zigbeeModel: ['929003074801_01', '929003074801_02', '929003074801_03'],
    model: '929001953101',
    vendor: 'Philips',
    description: 'Hue White and Color Ambiance GU10 (Xamento)',
    // Note that fromZigbee, toZigbee and exposes are missing here since we use extend here.
    // Extend contains a default set of fromZigbee/toZigbee converters and expose for common device types.
    // The following extends are available:
    // - extend.switch
    // - extend.light_onoff_brightness
    // - extend.light_onoff_brightness_colortemp
    // - extend.light_onoff_brightness_color
    // - extend.light_onoff_brightness_colortemp_color
    // extend: extend.light_onoff_brightness_colortemp_color(),
    extend: hueExtend.light_onoff_brightness_colortemp_color({colorTempRange: [153, 500]}),
    meta: {turnsOffAtBrightness1: true},
    ota: ota.zigbeeOTA,
};

module.exports = definition;
```

**Reply by ottoketney** [reply]
I have same behaviour. I moved back at the 1.28.4-1

**Reply by Koenkk** [reply]
External converters are not meant to be used for a long time (since they can break). Consider filling [an issue](https://github.com/Koenkk/zigbee2mqtt/issues/new?assignees=&labels=new+device+support&template=new_device_support.yaml&title=%5BNew+device+support%5D%3A+) to add support out-of-the-box.

**Reply by fireRoads** [reply]
Strange that the same code stopped working though. Any way, I filed an issue for my lights and they have already been committed. Looking forward to the next release, staying on 1.28.4-1 in the meantime. Thank you for your work!
